xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -119,8 +135,14 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
     lookasideChannelLb.handleResolvedAddresses(resolvedAddresses);
   }
 
+  /**
+   * Creates a balancer that is associated to a balancer name. The is to handle the legacy usecase
+   * that requires balancerName from xds config.
+   */
   private LoadBalancerProvider newLookasideChannelLbProvider(
       final String balancerName, final Node node, final List<ChannelCreds> channelCredsList) {
+    checkNotNull(balancerName, "balancerName is required");
```

Comment:
Is this really required? IMO, I would not bother have this.

Reply:
It was use to make sure the method is used only for the old usecase that needs balancer name.